app/background_tasks/refresh_course_task.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require 'rust_langs_cli_executor'
+require 'fileutils'
+
+class RefreshCourseTask
+  def initialize
+  end
+
+  def run
+    CourseTemplateRefresh.where(status: :not_started).each do |task|
+      channel_id = task.course_template_id
+      task.status = :in_progress
+      task.save!
+      ActionCable.server.broadcast("CourseTemplateRefreshChannel-#{channel_id}", { refresh_initialized: true })
+
+      courses = Course.where(course_template_id: task.course_template_id)
+      Rails.logger.info("Refreshing courses created from template #{task.course_template_id}")
+
+      rust_output = RustLangsCliExecutor.refresh(task.course_template, task.id)
+      task.langs_refresh_output = rust_output
+
+      broadcast_to_channel(channel_id, 'Updating database', 0.95, '-')
+      # Should we do this loop in CourseRefreshDatabaseUpdater?
+      courses.each do |course|
+        @refresh = CourseRefreshDatabaseUpdater.new.refresh_course(course, rust_output)
+      end
+
+      broadcast_to_channel(channel_id, 'Generating refresh report', 0.98, '-')
+      CourseTemplateRefreshReport.create(course_template_refresh_id: task.id, refresh_errors: @refresh.errors, refresh_warnings: @refresh.warnings, refresh_notices: @refresh.notices, refresh_timings: @refresh.timings)
+
+      broadcast_to_channel(channel_id, 'Cleaning up cache', 0.99, '-')
+      old_cache_path = task.course_template.cache_path
+      task.course_template.increment_cached_version
+      task.course_template.save!
```

Comment:
check that task.course_template matches rust_output['output-data'][new-cache-path]